### PR TITLE
Add package lock.json to VCS #7533

### LINKED ISSUE #7533: Add `package-lock.json`
Current behaviour: Everytime I run `npm install` this file, package-lock.json gets created and gets added to the git changes. 

Should this be ignored by git?

### PR DESCRIPTION
**Pull request by tshradheya**
Fixes #7533

**Outline of Solution**

1. Added `package-lock.json` to VCS after running `npm install` 
1. Updated doc to specify minimum Node.js version as `8.x`

### CODE REVIEW

**Comment by tshradheya** [comment]
Ready for Review

File: docs/dependencies.md [hunk 1]
```diff
@@ -25,4 +25,4 @@ If a library cannot be found in the NPM registry, simply host a local copy in th
 To update your local library configuration:
 
 - For production dependencies, only if you are adding/updating the library, find the files from the library that are necessary to be loaded to webpages, and add/update the entry(ies) in `FrontEndLibrary.java`.
-- For development dependencies, run `npm install` from the project root folder.
+- For development dependencies, run `npm install` from the project root folder and this will make changes made to the `package-lock.json` file which should be commited.
```
Comment (wkurniawan07) [inline docs/dependencies.md]: Let's rephrase a bit: "...from the project root folder. Additionally, if you are adding/updating the library, commit the changes made automatically to `package-lock.json`.".

Rationale: only if you're adding/updating the library then you will see the change.

Comment (tshradheya) [reply docs/dependencies.md]: Additionally, if you are adding/updating ~the~ any library, commit the changes made automatically to package-lock.json.

Updated by making the following change

**Comment by wkurniawan07** [comment]
@tshradheya For some reason my `package-lock.json` does not match yours. Instead of `"bundled": true` in about 100 places I have `resolved` and `integrity` fields. Can you check if deleting your `package-lock.json` and running `npm install` again still results in this copy of `package-lock.json` you committed?

**Comment by tshradheya** [comment]
Figured out why such a problem was there. 

`fsevents` is dealt differently in mac ( which is the system you probably have). My windows system ignored it and hence didnt install `fsevents` as it is optional. 

**Solution to fix**

- `npm install --no-optional` should be executed instead of just `npm install`.
- If it's still not installing optional dependecnies, try deleting `node_modules` and running the above command again. 

I am committing the new  `package-lock.json` which should be same as yours. And also updating the instructions in necessary places

**Comment by wkurniawan07** [comment]
In that case I don't think it's suitable for `package-lock.json` to be version-controlled. @whipermr5 any further thoughts?

**Comment by tshradheya** [comment]
Okay so researching about it a bit more, I realised that Node.js actually resolved this bug. [Source](http://blog.npmjs.org/post/167963735925/v560-2017-11-27)

However to make sure, it works perfectly fine for Teammates, I am going to try on mac and not mac to see if package-lock.jsom keeps changing due to optional dependencies. 
The solution will be to generate the package-lock file on mac system and check if on running npm install on other linux systems if the lock file gets updated or not( according to the bug fix it shouldn't) ( Our min requirement will then change to npm 5.6.0 which is automatically shipped with the latest Node.js 8.9.6)

Will probably need a day more to confirm everything and check if it is possible to commit it.

**Comment by wkurniawan07** [comment]
Let's close this for now. Can reopen when the Node.js community at large agrees that this `package-lock.json` should be included, which does not seem to be the case right now.

**Pull request closed without merging**